node/router/tests/connect.rs

```diff
@@ -150,11 +150,64 @@ async fn test_connect_with_handshake() {
     }
 }
 
+#[tokio::test]
+async fn test_validator_connection() {
+    // Create first router and start listening.
+    let node0 = validator(0, 2, &[], false).await;
+    assert_eq!(node0.number_of_connected_peers(), 0);
+    node0.enable_handshake().await;
+    node0.tcp().enable_listener().await.unwrap();
+
+    // Get the local IP address from the first router.
+    let addr0 = node0.local_ip();
+
+    // Create second router, trusting the first router, and start listening.
+    let node1 = validator(0, 2, &[addr0], false).await;
+    assert_eq!(node1.number_of_connected_peers(), 0);
+    node1.enable_handshake().await;
+    node1.tcp().enable_listener().await.unwrap();
+
+    {
+        // Connect node0 to node1.
+        node0.connect(node1.local_ip());
```

Comment:
aren't we racing against the `Heartbeat` (that might cause `node1` to attempt to connect to `node0`) here? then again, that direction should fail, so it's probably fine

Reply:
heartbeat is not turned on, only if we call `initialize_routing`.